index.d.ts

```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+declare module 'react-native-fontawesome' {
+  export const Icons: { [key: string]: string };
+  const Icon: React.ComponentClass<any>;
```

Comment:
This should be the same props as Text component I guess

Reply:
ah yeah, that's right! I wasn't thinking clearly.

Reply:
fixed!

Reply:
in visual studio code if i do

`import FontAwesome, { Icons } from 'react-native-fontawesome';`

and then i type **Icons.** autocomplete won't show up
it is not clearly defined which values icons can have here

``export const Icons: { [key: string]: string };``

Reply:
To do that there would need to be a type file with all the values that the `Icons` object has. 

I can do it, but it means that that type file needs to be updated every time that there are changes to the icons.

Reply:
I'll try to see if I can use some tool to generate the type definitions for the icons.

Reply:
ok, so the types can be generated automatically by renaming the icons file to `FontAwesomeIcons.ts` and then running `tsc FontAwesomeIcons.ts --declaration`.

Here's how it looks like:
https://gist.github.com/kristerkari/33821efa89e461d832b46660224e608e

The issue that I have currently is that I don't know how to import that declaration file without Typescript complaining.

Reply:
how about ?
```typescript
declare module 'react-native-fontawesome' {
    import React from 'react';
    export {default as Icons} from 'react-native-fontawesome/FontAwesomeIcons';

    const Icon: React.ComponentClass<any>;
    export default Icon;
}
```

Reply:
Hmm.. I think that I tried that already, but it did not work. Let me just try again to see if it works!

Reply:
for me it works, altho ts is 2.7.2

Reply:
yeah now that I tried again, that seems to work! I'll make a new PR.

Reply:
PR here:
https://github.com/entria/react-native-fontawesome/pull/32